src/core/allow-bind.c

```diff
@@ -0,0 +1,143 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#include "allow-bind.h"
+#include "unit.h"
+
+#if BUILD_BPF
+/* libbpf, clang and llc compile time dependencies are satisfied */
+#include "architecture.h"
+#include "bpf-object.h"
+#include "bpf/allow_bind/allow-bind-hexdump.h"
+#include "cgroup-bpf.h"
+#include "set.h"
+
+static int prepare_bpf_object(Unit* u, const Set *ports, struct bpf_object **ret_object) {
+        _cleanup_(bpf_object_freep) struct bpf_object *object = NULL;
```

Comment:
I think `bpf_object_free()` should be called when there is an error to be sure that `bpf_object__close()` is called. Am I missing something?

Reply:
this is what the `_cleanup_` macro does. It wraps gcc's cleanup concept, which allow declaring a function that is automaticlly invoked when a variable goes out of scope. we use it to reduce error codepath fuckups

Reply:
hmm, I don't know why I didn't see it.